src/libraries/System.Formats.Tar/src/System/Formats/Tar/TarHeader.Write.cs

```diff
@@ -1022,7 +1036,14 @@ private static int WriteChecksum(int checksum, Span<byte> buffer)
 
             Span<byte> converted = stackalloc byte[FieldLengths.Checksum];
             converted.Clear();
-            FormatOctal(checksum, converted);
+            FormatOctal(checksum, converted, shouldTrimTrailingNullBytes: true);
+
+            int length = converted.IndexOf((byte)'\0');
+            if (length != -1)
+            {
+                // Trim the nulls at the end
+                converted = converted.Slice(0, length);
+            }
```

Comment:
Won't converted be a right aligned octal number that is terminated by a `\0` byte?
If we need to ignore that byte, `j` can be initialized to `converted.Length - 2` below.

Reply:
I'm revisiting this code.

Reply:
> Won't converted be a right aligned octal number that is terminated by a \0 byte?

Yes. It's right aligned, padded with '0' chars on the left, and terminated with a null char.

I applied your suggestion and removed the boolean.